infrastructure/api/api.go

```diff
@@ -0,0 +1,180 @@
+package api
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/OrlandoRomo/academy-go-q32021/domain/model"
+)
+
+const (
+	urbanDictionaryURL = "https://mashape-community-urban-dictionary.p.rapidapi.com/define"
+	rapidapiKeyName    = "x-rapidapi-key"
+	csvPath            = "data/definitions.csv"
+)
+
+type UrbanDictionary struct {
+	ApiURL  string
+	Headers map[string]string
+	CSVPath string
+}
+
+// NewUrbanDictionary returns a new instance of the UrbanDictionary client
+func NewUrbanDictionary(apiKey string) *UrbanDictionary {
+	return &UrbanDictionary{
+		ApiURL: urbanDictionaryURL,
+		Headers: map[string]string{
+			rapidapiKeyName: apiKey,
+		},
+		CSVPath: csvPath,
+	}
+}
+
+func (u *UrbanDictionary) UpdateCSVPath(path string) {
+	u.CSVPath = path
+}
+
+// GetDefinitions reach out urban dictionary API by term
+func (u *UrbanDictionary) GetDefinitions(term string) (*model.List, error) {
+	var list *model.List
+	url := fmt.Sprintf("%s?term=%s", u.ApiURL, term)
+	request, err := http.NewRequest(http.MethodGet, url, nil)
```

Comment:
I suggest reading this article: https://www.thegreatcodeadventure.com/mocking-http-requests-in-golang/

It shows how you can reestructure a bit your code to be able to mock http calls, when doing tests you don't want to call the actual server